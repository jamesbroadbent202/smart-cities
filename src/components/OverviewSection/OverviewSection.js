import React from 'react';
import PropTypes from 'prop-types';
// import PageBanner from '../PageBanner/PageBanner';
import CategoryIndicator from '../CategoryIndicator/CategoryIndicator';
import {
  CATEGORIES,
  CATEGORY_IDS,
  // STRINGS,
} from '../../constants/index';
import getColorVariant from '../../helpers/getColorVariant';
import style from './OverviewSection.scss';

// const HERO_INDICATOR = 'population';

// TODO (davidg): delete this
const OverviewSection = (props) => {
  console.log('  --  >  OverviewSection.js:16 > OverviewSection');
  // const description = props.city
  //   ? props.city.description
  //   : STRINGS.OVERVIEW_DESCRIPTION;
  //
  // const title = props.city
  //   ? props.city.shortName
  //   : STRINGS.OVERVIEW_TITLE;

  return (
    <div className={style.wrapper}>
      {/* <PageBanner */}
      {/* colorName="primary" */}
      {/* description={description} */}
      {/* indicator={HERO_INDICATOR} */}
      {/* title={title} */}
      {/* cities={props.cities} */}
      {/* city={props.city} */}
      {/* /> */}

      <div className={style.indicatorCards}>
        {CATEGORIES
          .filter(category => category.id !== CATEGORY_IDS.CONTEXT)
          .map(category => (
            <div
              key={category.id}
              className={style.indicatorCardWrapper}
              style={{
                background: getColorVariant(category.colorName, '040'),
              }}
            >
              <CategoryIndicator
                className={style.categoryIndicator}
                city={props.city}
                cities={props.cities}
                category={category}
              />
            </div>
          ))}
      </div>
    </div>
  );
};

const cityPropShape = {
  description: PropTypes.string.isRequired,
  indices: PropTypes.object.isRequired,
  shortName: PropTypes.string.isRequired,
};

OverviewSection.propTypes = {
  city: PropTypes.shape(cityPropShape), // when on a city page
  cities: PropTypes.arrayOf(PropTypes.shape(cityPropShape)), // when on the all cities page
};

export default OverviewSection;
