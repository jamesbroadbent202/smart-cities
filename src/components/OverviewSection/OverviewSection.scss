@import '~style/variables';

.wrapper {
  position: relative;
  margin-bottom: 100px;
}

.indicatorCards {
  display: flex;
  flex-wrap: wrap;
  width: 100%;
}

.indicatorCardWrapper {
  display: flex;
  align-items: center;
  width: 100%;
  border-bottom: 4px solid $color-white;

  &:first-of-type {
    border-top: 4px solid $color-white;
  }

  @include for-tablet-landscape-up {
    width: 50%;

    &:first-of-type {
      border-top: none;
    }

    // left ones
    &:nth-of-type(odd) {
      justify-content: flex-end;
      border-right: 4px solid $color-white;
    }

    // first two
    &:nth-child(-n+2) {
      border-top: 4px solid $color-white;
    }
  }
}

.categoryIndicator {
  padding: 15px 16px 10px;
  width: 100%;
  border: none;

  @include for-tablet-landscape-up {
    padding: 25px 16px 10px;
  }

  @include for-tablet-landscape-up {
    // there's two side-by-side, so halve the usual container width
    // and allow for the fat border in the middle
    width: ($app-max-width / 2) - 2px;
    padding-left: 32px;
    padding-right: 32px;
  }

  @include for-desktop-up {
    // add more padding to the right of the left side
    // and left of the right side
    .indicatorCardWrapper:nth-of-type(odd) & {
      padding-right: 50px;
    }

    .indicatorCardWrapper:nth-of-type(even) & {
      padding-left: 50px;
    }
  }
}
