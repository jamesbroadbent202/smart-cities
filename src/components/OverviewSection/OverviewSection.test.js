import React from 'react';
import { shallow } from 'enzyme';
import OverviewSection from './OverviewSection';

jest.mock('../PageBanner/PageBanner', () => 'PageBanner');
jest.mock('../CategoryIndicator/CategoryIndicator', () => 'CategoryIndicator');
jest.mock('../../constants/index', () => ({
  CATEGORIES: [
    {
      id: 'cat-1',
      colorName: 'jobs',
    },
    {
      id: 'cat-2',
      colorName: 'cats',
    },
    {
      id: 'cat-3',
      colorName: 'dogs',
    },
  ],
  CATEGORY_IDS: {
    CONTEXT: 'context',
  },
  STRINGS: {
    OVERVIEW_DESCRIPTION: 'OVERVIEW_DESCRIPTION',
    OVERVIEW_TITLE: 'OVERVIEW_TITLE',
  }
}));

const defaultProps = {
  city: {
    description: 'The description',
    indices: {
      population: 77,
    },
    shortName: 'Perth',
  }
};

it('should match Snapshot', () => {
  const component = shallow(
    <OverviewSection {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should render each of the categories', () => {
  const component = shallow(
    <OverviewSection {...defaultProps} />
  );

  expect(component.find('CategoryIndicator').length).toBe(3);
});

describe('when a single city is passed in', () => {
  const component = shallow(
    <OverviewSection {...defaultProps} />
  );

  it('should show the city name', () => {
    expect(component.find('PageBanner').prop('title')).toBe('Perth');
  });

  it('should show the city description', () => {
    expect(component.find('PageBanner').prop('description')).toBe('The description');
  });
});

describe('when multiple cities are passed in', () => {
  // this would occur when on the all cities page
  const component = shallow(
    <OverviewSection
      {...defaultProps}
      city={undefined}
      cities={[
        {
          description: 'The description',
          indices: {
            population: 77,
          },
          shortName: 'Perth',
        },
      ]}
    />
  );

  it('should show the overview title', () => {
    expect(component.find('PageBanner').prop('title')).toBe('OVERVIEW_TITLE');
  });

  it('should show the overview description', () => {
    expect(component.find('PageBanner').prop('description')).toBe('OVERVIEW_DESCRIPTION');
  });
});
