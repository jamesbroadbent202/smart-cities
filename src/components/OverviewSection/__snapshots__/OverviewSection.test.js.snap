// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"wrapper\\">
  <PageBanner colorName=\\"primary\\" description=\\"The description\\" indicator=\\"population\\" title=\\"Perth\\" cities={[undefined]} city={{...}} />
  <div className=\\"indicatorCards\\">
    <div className=\\"indicatorCardWrapper\\" style={{...}}>
      <CategoryIndicator className=\\"categoryIndicator\\" city={{...}} cities={[undefined]} category={{...}} />
    </div>
    <div className=\\"indicatorCardWrapper\\" style={{...}}>
      <CategoryIndicator className=\\"categoryIndicator\\" city={{...}} cities={[undefined]} category={{...}} />
    </div>
    <div className=\\"indicatorCardWrapper\\" style={{...}}>
      <CategoryIndicator className=\\"categoryIndicator\\" city={{...}} cities={[undefined]} category={{...}} />
    </div>
  </div>
</div>"
`;
